overlord/state/notices.go

```diff
@@ -127,19 +207,22 @@ type jsonNotice struct {
 	LastData      map[string]string `json:"last-data,omitempty"`
 	RepeatAfter   string            `json:"repeat-after,omitempty"`
 	ExpireAfter   string            `json:"expire-after,omitempty"`
+
+	RepeatCheckData *json.RawMessage `json:"repeat-check-data,omitempty"`
 }
 
 func (n *Notice) MarshalJSON() ([]byte, error) {
-	jn := jsonNotice{
-		ID:            n.id,
-		UserID:        n.userID,
-		Type:          string(n.noticeType),
-		Key:           n.key,
-		FirstOccurred: n.firstOccurred,
-		LastOccurred:  n.lastOccurred,
-		LastRepeated:  n.lastRepeated,
-		Occurrences:   n.occurrences,
-		LastData:      n.lastData,
+	jn := marshalledNotice{
```

Comment:
Can we not define `MarshalJSON` / `UnmarshalJSON` now that these should only be used internally to the state package with the `RepeatCheckData` field in there? (But for example just convert to `jsonNotice` when we need to marshal?)

Reply:
I followed the pattern for marshaling/unmarshaling `Change` and `Task`. I not sure, I am not sure I fully understand, do you mean converting to/from `jsonNotice` in `state.{MarshalJSON,UnmarshalJSON}` directly?